pages/maps/index.vue

```diff
@@ -45,12 +43,60 @@ export default {
           },
           label: 'Home'
         }
-      ]
+      ],
+      uuid: undefined,
+      prefix: "/maps",
+      state: undefined,
+      api: process.env.portal_api,
+      flatmapAPI: process.env.flatmap_api,
+    }
+  },
+  computed: {
+    shareLink: function() {
+      if (this.uuid)
+        return this.prefix +"?id=" + this.uuid ;
+      return this.prefix;
     }
   },
+  methods: {
+    updateUUID: function() {
+      let xmlhttp = new XMLHttpRequest();
```

Comment:
Nuxt has a [fetch hook](https://nuxtjs.org/docs/2.x/components-glossary/pages-fetch) that can be used in place of XHR. This would allow this code to be less lines and more efficient. Not a necessary changed to be made, as all routes were accounted for here.

Reply:
Thanks for point that out. I will make a new pull request with the suggested changes.